validator_client/src/duties_service.rs

```diff
@@ -570,26 +571,44 @@ async fn poll_beacon_attesters<T: SlotClock + 'static, E: EthSpec>(
             });
     }
 
-    // If there are any subscriptions, push them out to the beacon node.
+    // If there are any subscriptions, push them out to beacon nodes
     if !subscriptions.is_empty() {
         let subscriptions_ref = &subscriptions;
-        if let Err(e) = duties_service
-            .beacon_nodes
-            .first_success(
-                duties_service.require_synced,
-                OfflineOnFailure::Yes,
-                |beacon_node| async move {
-                    let _timer = metrics::start_timer_vec(
-                        &metrics::DUTIES_SERVICE_TIMES,
-                        &[metrics::SUBSCRIPTIONS_HTTP_POST],
-                    );
-                    beacon_node
-                        .post_validator_beacon_committee_subscriptions(subscriptions_ref)
-                        .await
-                },
-            )
-            .await
-        {
+        if let Err(e) = if duties_service.disable_run_on_all {
```

Comment:
What're your thoughts on adding a really simple function to `BeaconNodeFallback` so we don't have to have these `if` `else` blocks?

```
    pub async fn run<'a, F, Err, R>(
        &'a self,
        require_synced: RequireSynced,
        offline_on_failure: OfflineOnFailure,
        func: F,
        disable_run_on_all: bool,
    ) -> Result<(), Errors<Err>>
        where
            F: Fn(&'a BeaconNodeHttpClient) -> R,
            R: Future<Output = Result<(), Err>>,
    {
        if disable_run_on_all {
            self.first_success(require_synced, offline_on_failure, func).await?;
            Ok(())
        } else {
            self.run_on_all(require_synced, offline_on_failure, func).await
        }
    }
```

We could also add the `disable_run_on_all` bool to `BeaconNodeFallback` so we don't have to pass it in as an argument

Reply:
I like this idea. Will make the changes.